### PR TITLE
[CARBONDATA-846] Added support to revert changes if query fails

### PR DESCRIPTION
**Pull request by kunal642**

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1412/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1417/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1422/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1423/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/AlterTableCommands.scala [hunk 1]
```diff
@@ -89,9 +89,8 @@ private[sql] case class AlterTableAddColumns(
       LOGGER.info(s"Alter table for add columns is successful for table $dbName.$tableName")
       LOGGER.audit(s"Alter table for add columns is successful for table $dbName.$tableName")
     } catch {
-      case e: Exception =>
-        LOGGER.error(e, s"Alter table add columns failed : ${e.getMessage}")
-        // clean up the dictionary files in case of any failure
+      case runTimeException: RuntimeException => LOGGER
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/AlterTableCommands.scala]: why catch only `RuntimeException` ? why not `Exception` or any specific exception.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1473/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1474/

File: integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala [hunk 2]
```diff
@@ -96,4 +104,109 @@ object AlterTableUtil {
     }
     schemaParts.mkString(",")
   }
+
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala]: Add comments on all methods

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1478/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1479/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1481/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1485/

**Comment by manishgupta88** [comment]
@kunal642 change column name to column ID for every check in the revert methods

**Comment by kunal642** [comment]
@manishgupta88 ok

**Comment by manishgupta88** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1486/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1487/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1489/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**